### PR TITLE
rational class

### PR DESCRIPTION
**Pull request by alsiola**

### CODE REVIEW

File: src/main/scala/Rational.scala [hunk 1]
```diff
@@ -0,0 +1,65 @@
+case class Rational(n: Int, d: Int) {
+  require(d != 0)
```
Comment (jSherz) [inline src/main/scala/Rational.scala]: I think it's nice to add in a message for assertions to give the user feedback: https://www.scala-lang.org/api/current/scala/Predef$.html#require(requirement:Boolean,message:=%3EAny):Unit

File: src/main/scala/Rational.scala [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  val g = gcd(n.abs, d.abs);
+  val numerator = n / g
+  val denominator = d / g
+
+  override def toString(): String = s"Rational($numerator/$denominator)"
+  override def equals(obj: Any): Boolean =
+    obj match {
```
Comment (jSherz) [inline src/main/scala/Rational.scala:11]: You can sometimes replace match statements with an anonymous function:

```scala
  override def equals(obj: Any): Boolean = {
    case r: Rational =>
      numerator == r.numerator && denominator == r.denominator
    case _ => false
  }
```

Comment (jSherz) [reply src/main/scala/Rational.scala:11]: (not sure if that syntax is quite correct but it'd be similar)

**Review by jSherz — APPROVED** [review]
NOICE. You may want to add some gitignores for those top files and also:

https://scalameta.org/scalafmt/
http://www.scalastyle.org/

For tests:

https://www.scalatest.org/

**Comment by alsiola** [comment]
> For tests:

what are tests and why would I want them?

**Pull request merged**